src/block-walker.ts

```diff
@@ -0,0 +1,214 @@
+import { BMT_IDs } from "./common-types";
+import { IDbWrapper, MasterChainBlockInfo } from "./db-wrapper";
+import { RangerConfig } from "./ranger-config";
+
+export class BlockWalker {
+    static MAX_DEPTH = 10;
+    readonly buffer_db: IDbWrapper;
+    readonly ordered_db: IDbWrapper;
+    readonly masterchain_block_id: string;
+
+    masterchain_block_info?: MasterChainBlockInfo;
```

Comment:
It is not good to declare the majority of the fields as an optional and dereference it with `!`. You have to find better class design.